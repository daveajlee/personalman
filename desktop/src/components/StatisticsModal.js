import {Button, Modal} from "react-bootstrap";
import React, {useEffect, useState} from "react";
import axios from "axios";

function StatisticsModal (props) {

    const [statisticsMap, setStatisticsMap] = useState([]);
    const [leaveEntitlement, setLeaveEntitlement] = useState(0);

    const handleStatisticsClose = () => props.setShowStatisticsModal(false);

    useEffect(() => {
        // Load the statistics for the current year.
        let startYearDate = '01-01-' + props.year;
        let endYearDate = '31-12-' + props.year;
        axios.get('http://localhost:8150/api/absences/?company=' + props.company + '&username=' + props.token.split("-")[0] + '&startDate=' + startYearDate + '&endDate=' + endYearDate + '&onlyCount=false&token=' + props.token)
            .then(res => {
                const result = res.data;
                setStatisticsMap(result['statisticsMap']);
            })
        // Get the leave entitlement for this user.
        axios.get('http://localhost:8150/api/user/?company=' + props.company + '&username=' + props.token.split("-")[0] + '&token=' + props.token)
            .then(res => {
                const result = res.data;
                setLeaveEntitlement(result['leaveEntitlementPerYear']);
            })
    }, [props.company, props.token, props.year]);

    return (
        <Modal show={props.showStatisticsModal} onHide={props.handleStatisticsClose}>
            <Modal.Header closeButton>
                <Modal.Title>View Statistics - {props.username} - {props.year}</Modal.Title>
            </Modal.Header>
            <Modal.Body>Illness: {statisticsMap['Illness']} days <br/> Holiday: {statisticsMap['Holiday']} days (Remaining: {leaveEntitlement} days) <br/>
                Trip: {statisticsMap['Trip']} days <br/> Conference: {statisticsMap['Conference']} days <br/> Day in Lieu: {statisticsMap['Day in Lieu']} days (Remaining: {statisticsMap['Day in Lieu Request']} days) <br/>
                Federal Holiday: {statisticsMap['Federal Holiday']} days</Modal.Body>
            <Modal.Footer>
                <Button variant="primary" onClick={handleStatisticsClose}>
                    Close
                </Button>
            </Modal.Footer>
        </Modal>
    );

}

export default StatisticsModal;
