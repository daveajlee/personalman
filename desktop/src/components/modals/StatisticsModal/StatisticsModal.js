import {Button, Modal} from "react-bootstrap";
import React, {useEffect, useState} from "react";
import axios from "axios";
import PropTypes from "prop-types";
import {useTranslation} from "react-i18next";

/**
 * This is the modal window to show the view statistics for a particular user. The modal is used for admin users who want
 * to view the statistics of a particular user.
 * @param props setShowStatisticsModal - function to determine whether modal should be shown or not,
 * company - the company that the user belongs to, username - the username of the user that the statistics should be displayed for,
 * token - the current user access token of the admin user, year - the year that the statistics should be displayed for
 * @returns {JSX.Element} to be displayed to the user.
 */
function StatisticsModal (props) {

    const [statisticsMap, setStatisticsMap] = useState([]);
    const [leaveEntitlement, setLeaveEntitlement] = useState(0);

    const {t} = useTranslation();

    /**
     * Function to handle the case that the user clicks on the close button in the modal.
     */
    const handleStatisticsClose = () => props.setShowStatisticsModal(false);

    /**
     * Load the statistics from the REST API based on the supplied information.
     */
    useEffect(() => {
        // Load the statistics for the current year.
        let startYearDate = '01-01-' + props.year;
        let endYearDate = '31-12-' + props.year;
        let username;
        if ( props.username === '') {
            username = props.token.split("-")[0];
        } else {
            username = props.username;
        }
        axios.get(process.env.REACT_APP_SERVER_URL + '/absences/?company=' + props.company + '&username=' + username + '&startDate=' + startYearDate + '&endDate=' + endYearDate + '&onlyCount=false&token=' + props.token)
            .then(res => {
                const result = res.data;
                setStatisticsMap(result['statisticsMap']);
            }).catch(error => {
                console.error(error);
        })
        // Get the leave entitlement for this user.
        axios.get(process.env.REACT_APP_SERVER_URL  + '/user/?company=' + props.company + '&username=' + username + '&token=' + props.token)
            .then(res => {
                const result = res.data;
                setLeaveEntitlement(result['leaveEntitlementPerYear']);
            }).catch(error => {
                console.error(error);
        })
    }, [props.company, props.token, props.year, props.username]);

    /**
     * Display the relevant elements and data to the user.
     */
    return (
        <Modal show={props.showStatisticsModal} onHide={handleStatisticsClose}>
            <Modal.Header closeButton>
                <Modal.Title>{t('statisticsModalTitle')} - {props.username} - {props.year}</Modal.Title>
            </Modal.Header>
            <Modal.Body>{t('Illness')}: {statisticsMap['Illness']} {t('statisticsModalDays')} <br/> {t('Holiday')}: {statisticsMap['Holiday']} {t('statisticsModalDays')} ({t('statisticsModalRemaining')}: {leaveEntitlement} {t('statisticsModalDays')}) <br/>
                {t('Trip')}: {statisticsMap['Trip']} {t('statisticsModalDays')} <br/> {t('Conference')}: {statisticsMap['Conference']} {t('statisticsModalDays')} <br/> {t('DayinLieu')}: {statisticsMap['Day in Lieu']} {t('statisticsModalDays')} ({t('statisticsModalRemaining')}: {statisticsMap['Day in Lieu Request']} {t('statisticsModalDays')}) <br/>
                {t('FederalHoliday')}: {statisticsMap['Federal Holiday']} {t('statisticsModalDays')}</Modal.Body>
            <Modal.Footer>
                <Button variant="primary" onClick={handleStatisticsClose}>
                    Close
                </Button>
            </Modal.Footer>
        </Modal>
    );

}

export default StatisticsModal;

StatisticsModal.propTypes = {
    /**
     * the company that the user belongs to
     */
    company: PropTypes.string,

    /**
     * function to determine whether modal should be shown or not
     */
    setShowStatisticsModal: PropTypes.func,

    /**
     * the username of the user that the statistics should be displayed for
     */
    username: PropTypes.string,

    /**
     * the current user access token of the admin user
     */
    token: PropTypes.string,

    /**
     * the year that the statistics should be displayed for
     */
    year: PropTypes.number
}

    StatisticsModal.defaultProps = {
    company: 'Required',
    username: 'Required',
    token: 'Required'
};
