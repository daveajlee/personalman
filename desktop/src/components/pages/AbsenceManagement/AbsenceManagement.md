```jsx
import { HashRouter } from "react-router-dom";

<HashRouter>
    <AbsenceManagement docMode="true"/>
</HashRouter>
```
