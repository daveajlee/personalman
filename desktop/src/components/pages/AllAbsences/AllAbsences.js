import React, {useEffect} from "react";
import {useLocation} from "react-router-dom";
import {Container} from "react-bootstrap";
import Header from "../../layout/Header/Header";
import {useState} from "react";
import axios from "axios";
import AbsenceList from "../../lists/AbsenceList/AbsenceList";

/**
 * This is the page which displays the list of absences for all users for an admin user within a particular company.
 * @returns {JSX.Element} to be displayed to the user.
 */
function AllAbsences(props) {

    const location = useLocation();

    const [startDate, setStartDate] = useState("");
    const [endDate, setEndDate] = useState("");
    const [role, setRole] = useState("");

    /**
     * Retrieve the current token either from the supplied state or empty if we are in doc mode.
     */
    const token = (props.docMode && props.docMode==='true') ? "" : location.state.token;

    /**
     * Retrieve the company either from the supplied state or empty if we are in doc mode.
     */
    const company = ( props.docMode && props.docMode==='true') ? "" : location.state.company;

    /**
     * Retrieve the year either from the supplied state or empty if we are in doc mode.
     */
    const year = ( props.docMode && props.docMode==='true') ? "" :  location.state.year ? location.state.year : new Date().getFullYear();

    /**
     * Retrieve the month either from the supplied state or empty if we are in doc mode.
     */
    const month = ( props.docMode && props.docMode==='true') ? "" : location.state.month ? location.state.month : ((new Date().getMonth()) +1);

    /**
     * Load the absences from the REST API either for all users within a specific time period
     * (usually a month in a year).
     */
    useEffect(() => {
        // Ensure that the user is actually admin, otherwise they cannot view the page.
        axios.get(process.env.REACT_APP_SERVER_URL + '/user/?company=' + company + '&username=' + token.split("-")[0] + '&token=' + token)
                .then(res => {
                    const result = res.data;
                    setRole(result['role']);
                }).catch(error => {
                    console.error(error);
        })
        // Load the actual month of absences.
        /*if ( month === null ) {
            setStartDate('01-' + ((new Date().month) +1).toLocaleString('en-US', {
                minimumIntegerDigits: 2,
                useGrouping: false
            }) + '-' + new Date().getFullYear());
            setEndDate(daysInMonth(((new Date().month) +1), new Date().getFullYear()) + '-' + ((new Date().month) +1).toLocaleString('en-US', {
                minimumIntegerDigits: 2,
                useGrouping: false
            }) + '-' + new Date().getFullYear());
        } else {*/
            setStartDate('01-' + (month).toLocaleString('en-US', {
                minimumIntegerDigits: 2,
                useGrouping: false
            }) + '-' + year);
            setEndDate(daysInMonth(month, year) + '-' + (month).toLocaleString('en-US', {
                minimumIntegerDigits: 2,
                useGrouping: false
            }) + '-' + year);
        //}

    }, [token, month, year, company]);

    /**
     * Return the number of days that a month has in a particular year.
     * @param month the month to retrieve the number of days for.
     * @param year the year that should be taken into account.
     * @returns {number} the number of days that the month has in the supplied year.
     */
    function daysInMonth (month, year) {
        return new Date(year, month, 0).getDate();
    }

    /**
     * Display the relevant elements and data to the user.
     */
    return (
        <Container fluid>
            <Header token={token} company={company}/>

            { role==='Admin' && <AbsenceList company={company} token={token} username='' startDate={startDate} endDate={endDate}
                         month={month}
                         year={year}/>}

        </Container>
    )


}

export default AllAbsences;
