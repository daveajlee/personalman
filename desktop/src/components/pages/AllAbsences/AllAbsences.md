```jsx
import { HashRouter } from "react-router-dom";

<HashRouter>
    <AllAbsences docMode="true"/>
</HashRouter>
```
