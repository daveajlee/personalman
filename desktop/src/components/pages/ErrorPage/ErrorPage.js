import React from "react";
import {Col, Container, Image, Row} from "react-bootstrap";
import logo from './../../../assets/personalman-logo.png';

/**
 * This page is the error page which is displayed whenever the user visits an invalid url.
 * @returns {JSX.Element} to be displayed to the user.
 */
function ErrorPage() {

    /**
     * Display the relevant elements and data to the user.
     */
    return (
        <Container fluid className="p-3 my-5 h-custom">
            <Row className="d-flex flex-row align-items-center justify-content-center">
                <Col>
                    <Image src={logo}
                           className="d-block mx-auto img-fluid w-50" alt="PersonalMan Logo"/>
                </Col>
                <Col>
                    <Container className="d-flex flex-row align-items-center justify-content-center mb-3">
                        <Row>
                            <Col>
                                <h1>Sorry this page could not be found!</h1>

                                <h2><a href={"/"}>Click here</a> to return to the home page.</h2>
                            </Col>
                        </Row>
                    </Container>
                </Col>
            </Row>
        </Container>
    );

}

export default ErrorPage;
