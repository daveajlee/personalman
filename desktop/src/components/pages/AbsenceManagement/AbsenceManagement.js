import React, {useLayoutEffect} from "react";
import {useLocation} from "react-router-dom";
import {Container, Row, Col, Button, Modal, Form} from "react-bootstrap";
import Header from "../../layout/Header/Header";
import {useState} from "react";
import axios from "axios";
import StatisticsModal from "../../modals/StatisticsModal/StatisticsModal";
import AbsenceList from "../../lists/AbsenceList/AbsenceList";
import {useTranslation} from "react-i18next";

/**
 * This is the page which displays the list of absences for the current user and allows them to create new absences or
 * view other months or add an absence.
 * @returns {JSX.Element} to be displayed to the user.
 */
function AbsenceManagement(props) {

    const location = useLocation();

    const [showAddModal, setShowAddModal] = useState(false);
    const [showStatisticsModal, setShowStatisticsModal] = useState(false);

    const {t} = useTranslation();

    /**
     * Function to manage the modal for adding absences.
     */
    const handleAddClose = () => setShowAddModal(false);

    const [startDate, setStartDate] = useState("");
    const [endDate, setEndDate] = useState("");
    const [reason, setReason] = useState("Illness");

    /**
     * Retrieve the current token either from the supplied state or empty if we are in doc mode.
     */
    const token = (props.docMode && props.docMode==='true') ? "" : location.state.token;

    /**
     * Retrieve the company either from the supplied state or empty if we are in doc mode.
     */
    const company = ( props.docMode && props.docMode==='true') ? "" : location.state.company;

    /**
     * Retrieve the year either from the supplied state or empty if we are in doc mode. 
     */
    const year = ( props.docMode && props.docMode==='true') ? "" :  location.state.year ? location.state.year : new Date().getFullYear();
    
    /**
     * Retrieve the month either from the supplied state or empty if we are in doc mode.
     */
    const month = ( props.docMode && props.docMode==='true') ? "" : location.state.month ? location.state.month : ((new Date().getMonth()) +1);

    /**
     * Load the absences from the REST API either for a particular username within a specific time period
     * (usually a month in a year).
     */
    useLayoutEffect(() => {
        // Load the actual month of absences.
        /*if ( !location.state.month ) {
            setStartDate('01-' + ((new Date().month) +1).toLocaleString('en-US', {
                minimumIntegerDigits: 2,
                useGrouping: false
            }) + '-' + new Date().getFullYear());
            setEndDate(daysInMonth(((new Date().month) +1), new Date().getFullYear()) + '-' + ((new Date().month) +1).toLocaleString('en-US', {
                minimumIntegerDigits: 2,
                useGrouping: false
            }) + '-' + new Date().getFullYear());
        } else {*/
            setStartDate('01-' + (month).toLocaleString('en-US', {
                minimumIntegerDigits: 2,
                useGrouping: false
            }) + '-' + year);
            setEndDate(daysInMonth(month, year) + '-' + (month).toLocaleString('en-US', {
                minimumIntegerDigits: 2,
                useGrouping: false
            }) + '-' + year);
        //}

    }, [token, month, year, company]);

    /**
     * Return the number of days that a month has in a particular year.
     * @param month the month to retrieve the number of days for.
     * @param year the year that should be taken into account.
     * @returns {number} the number of days that the month has in the supplied year.
     */
    function daysInMonth (month, year) {
        return new Date(year, month, 0).getDate();
    }

    /**
     * Get the username of the currently logged in user who we will display absences for.
     * @returns the username
     */
    function getUsername() {
        return token.split("-")[0];
    }

    /**
     * Calculate the statistics of the user such as holidays etc.
     */
    function viewStatistics() {
        setShowStatisticsModal(true);
    }

    /**
     * Show the input modal to add an absence.
     */
    function addAbsenceInput() {
        setShowAddModal(true);
    }

    /**
     * Add the actual absence.
     */
    function addAbsence() {
        let startDateSplit = startDate.split("-");
        let endDateSplit = endDate.split("-");
        if ( endDateSplit[2] < startDateSplit[2] && endDateSplit[1] <= startDateSplit[1] && endDateSplit[0] <= startDateSplit[0] ) {
            alert(t('absenceManagementAddAbsenceFail'));
            return;
        }
        axios.post(process.env.REACT_APP_SERVER_URL + '/absences/', {
            company: company,
            username: getUsername(),
            startDate: startDateSplit[2] + '-' + startDateSplit[1] + '-' + startDateSplit[0],
            endDate: endDateSplit[2] + '-' + endDateSplit[1] + '-' + endDateSplit[0],
            category: reason,
            token: token,
        }).then((response) => {
            if ( response.status === 201 ) {
                alert(t('absenceManagementAddAbsenceSuccess'));
                setShowAddModal(false);
                window.location.reload();
            }
        }).catch((error) => {
            console.error(error);
        });
    }

    /**
     * Set the start date that the user entered to the state for later.
     * @param event the event triggered by the user.
     */
    function startDateChangeHandler(event) {
        setStartDate(event.target.value);
    }

    /**
     * Set the end date that the user entered to the state for later.
     * @param event the event triggered by the user.
     */
    function endDateChangeHandler(event) {
        setEndDate(event.target.value);
    }

    /**
     * Set the reason that the user entered to the state for later.
     * @param event the event triggered by the user.
     */
    function reasonChangeHandler(event) {
        setReason(event.target.value);
    }

    /**
     * Display the relevant elements and data to the user.
     */
    return (
        <Container fluid>
            <Header token={token} company={company}/>

            <AbsenceList company={company} token={token} startDate={startDate} endDate={endDate}
            month={month}
            year={year}
            username={getUsername()}/>

            <Container className='align-items-center justify-content-center text-md-start mt-4 pt-2'>
                <Row>
                    <Col className="text-center">
                        <Button className="mb-0 px-5 me-2" size='lg' onClick={addAbsenceInput}>{t('absenceManagementAddAbsence')}</Button>
                        <Button className="mb-0 px-5 me-2" size='lg' onClick={viewStatistics}>{t('absenceManagementViewStatistics')}</Button>
                    </Col>
                </Row>
            </Container>

            <Modal show={showAddModal} onHide={handleAddClose}>
                <Modal.Header closeButton>
                    <Modal.Title>{t('absenceManagementAddAbsenceModal')}</Modal.Title>
                </Modal.Header>
                <Modal.Body>
                    <Form.Group as={Row} className="mb-3" controlId="formPlaintextStartDate">
                        <Form.Label column sm="2">{t('absenceManagementStartDateModal')}:</Form.Label>
                        <Col sm="10">
                            <Form.Control type="date" value={startDate} placeholder="yyyy-MM-dd" onChange={startDateChangeHandler}/>
                        </Col>
                    </Form.Group>
                    <Form.Group as={Row} className="mb-3" controlId="formPlaintextEndDate">
                        <Form.Label column sm="2">{t('absenceManagementEndDateModal')}:</Form.Label>
                        <Col sm="10">
                            <Form.Control type="date" value={endDate} placeholder="yyyy-MM-dd" onChange={endDateChangeHandler}/>
                        </Col>
                    </Form.Group>
                    <Form.Group as={Row} className="mb-3" controlId="formPlaintextReason">
                        <Form.Label column sm="2">{t('absenceManagementReasonModal')}:</Form.Label>
                        <Col sm="10">
                            <Form.Select aria-label="Role" value={reason} onChange={reasonChangeHandler}>
                                <option key="illness">{t('Illness')}</option>
                                <option key="holiday">{t('Holiday')}</option>
                                <option key="trip">{t('Trip')}</option>
                                <option key="conference">{t('Conference')}</option>
                                <option key="dayinlieu">{t('DayinLieu')}</option>
                                <option key="federalholiday">{t('FederalHoliday')}</option>
                            </Form.Select>
                        </Col>
                    </Form.Group>
                </Modal.Body>
                <Modal.Footer>
                    <Button variant="secondary" onClick={handleAddClose}>
                        {t('cancel')}
                    </Button>
                    <Button variant="primary" onClick={addAbsence}>
                        {t('absenceManagementAddAbsence')}
                    </Button>
                </Modal.Footer>
            </Modal>

            <StatisticsModal year={ year} company={company}
                             token={token} showStatisticsModal={showStatisticsModal} setShowStatisticsModal={setShowStatisticsModal}
                             username={getUsername()}>
            </StatisticsModal>

        </Container>
    )

}

export default AbsenceManagement;
