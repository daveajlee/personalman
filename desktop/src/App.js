import 'bootstrap/dist/css/bootstrap.min.css';
import './css/App.css'
import Login from "./components/pages/Login/Login";
import RegisterCompany from "./components/pages/RegisterCompany/RegisterCompany";
import React, {useEffect, useState} from "react";
import axios from "axios";
import {Route, Routes} from "react-router-dom";
import RegisterUser from "./components/pages/RegisterUser/RegisterUser";
import AbsenceManagement from "./components/pages/AbsenceManagement/AbsenceManagement";
import ChangePassword from "./components/pages/ChangePassword/ChangePassword";
import UsersManagement from "./components/pages/UsersManagement/UsersManagement";
import AllAbsences from "./components/pages/AllAbsences/AllAbsences";
import Logout from "./components/pages/Logout/Logout";
import './i18n/config';

/**
 * This is the first page that the user visits when starting PersonalMan - either show the login screen or the register
 * a company screen if no companies have been defined so far.
 * @returns {JSX.Element} to be displayed to the user.
 */
function App() {

  const [companies, setCompanies] = useState([]);
  const [loading, setLoading] = useState(true);

    /**
     * Load the current list of companies that are available on the REST API server.
     */
  useEffect(() => {
    axios.get(process.env.REACT_APP_SERVER_URL + `/companies/`)
        .then(res => {
          setLoading(false);
          const companies = res.data;
          setCompanies(companies);
        }).catch(error => {
            console.log('No connection to API');
    })
  }, []);

    /**
     * Display the relevant elements and data to the user.
     */
    return (
        <Routes>
            <Route exact path="/" element={!loading && companies && companies.length > 0 ? <Login/> : <RegisterCompany/>}/>
            <Route path='/login' element={<Login/>} />
            <Route path='/registerCompany' element={<RegisterCompany/>} />
            <Route path='/registerUser' element={<RegisterUser/>} />
            <Route path='/absences' element={<AbsenceManagement/>} />
            <Route path='/changePassword' element={<ChangePassword/>} />
            <Route path='/users' element={<UsersManagement/>} />
            <Route path='/allAbsences' element={<AllAbsences/>} />
            <Route path='/logout' element={<Logout/>} />
        </Routes>
  );
}

export default App;
