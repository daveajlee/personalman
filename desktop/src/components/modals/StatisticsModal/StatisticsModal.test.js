import renderer from 'react-test-renderer';
import {createBrowserRouter, RouterProvider} from "react-router-dom";
import React from "react";
import AbsenceManagement from "../../pages/AbsenceManagement/AbsenceManagement";
import AllAbsences from "../../pages/AllAbsences/AllAbsences";
import App from "../../../App";
import ErrorPage from "../../pages/ErrorPage/ErrorPage";
import StatisticsModal from "./StatisticsModal";

it('checks that the StatisticsModal component can be rendered', () => {

    const component = renderer.create(
        <React.StrictMode>
            <RouterProvider router={createBrowserRouter([{
                path: "*",
                element: <App/>,
                errorElement: <ErrorPage />,
            },{
                path: "/absences",
                element: <AbsenceManagement/>
            },{
                path: "/allAbsences",
                element: <AllAbsences/>
            }])}>
            <StatisticsModal company="Mustermann GmbH" setShowStatisticsModal={() => {}} username="mmustermann" token="mmustermann-ghgkg" year={2013}/>
            </RouterProvider>
        </React.StrictMode>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
