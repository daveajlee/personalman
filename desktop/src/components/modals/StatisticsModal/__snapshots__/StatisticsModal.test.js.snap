// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checks that the StatisticsModal component can be rendered 1`] = `
<div
  className="p-3 my-5 h-custom container-fluid"
>
  <div
    className="d-flex flex-row align-items-center justify-content-center row"
  >
    <div
      className="col"
    >
      <img
        alt="PersonalMan Logo"
        className="d-block mx-auto img-fluid w-50"
        src="https://www.davelee.de/common/assets/img/portfolio/personalman-logo.png"
      />
    </div>
    <div
      className="col"
    >
      <div
        className="d-flex flex-row align-items-center justify-content-center mb-3 container"
      >
        <div
          className="row"
        >
          <div
            className="col"
          >
            <h3>
              registerCompanyTitle
            </h3>
          </div>
        </div>
      </div>
      <div
        className="d-flex flex-column container"
      >
        <form
          className=""
        >
          <div
            className="mb-3 row"
          >
            <label
              className="form-label col-form-label col-sm-2"
              htmlFor="formPlaintextName"
            >
              registerCompanyNameLabel
              :
            </label>
            <div
              className="col-sm-10"
            >
              <input
                className="form-control"
                id="formPlaintextName"
                onChange={[Function]}
                placeholder="registerCompanyNameLabel"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            className="mb-3 row"
          >
            <label
              className="form-label col-form-label col-sm-2"
              htmlFor="formPlaintextEmail"
            >
              registerCompanyAnnualLeaveLabel
              :
            </label>
            <div
              className="col-sm-10"
            >
              <input
                className="form-control"
                id="formPlaintextEmail"
                max="40"
                min="20"
                onChange={[Function]}
                type="number"
                value={20}
              />
            </div>
          </div>
          <div
            className="mb-3 row"
          >
            <label
              className="form-label col-form-label col-sm-2"
              htmlFor="formPlaintextCompany"
            >
              registerCompanyBaseCountryLabel
              :
            </label>
            <div
              className="col-sm-10"
            >
              <select
                aria-label="Company Name"
                className="form-select"
                id="formPlaintextCompany"
                onChange={[Function]}
                value="Germany"
              >
                <option>
                  registerCompanyCountryGermany
                </option>
                <option>
                  registerCompanyCountryEngland
                </option>
                <option>
                  registerCompanyCountryScotland
                </option>
              </select>
            </div>
          </div>
        </form>
      </div>
      <div
        className="align-items-center justify-content-center text-md-start mt-4 pt-2 container"
      >
        <div
          className="row"
        >
          <div
            className="text-center col"
          >
            <button
              className="mb-0 px-5 me-2 btn btn-primary btn-lg"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              registerCompanyRegisterButton
            </button>
            <button
              className="mb-0 px-5 me-2 btn btn-primary btn-lg"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              registerCompanyResetButton
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
